File: app/src/main/java/com/example/pam_project/database/lists/ListMapper.java
package com.example.pam_project.database.lists;

import com.example.pam_project.database.relationships.ListsWithTasks;
import com.example.pam_project.database.tasks.TaskEntity;
import com.example.pam_project.features.lists.list.ListInformation;
import com.example.pam_project.features.tasks.list.TaskInformation;
import com.example.pam_project.utils.TaskStatus;

import java.util.ArrayList;
import java.util.List;

public class ListMapper {

    public ListInformation toModel(final ListEntity entity) {
        return new ListInformation(entity.id, entity.name, entity.categoryId);
    }

    public ListInformation toModelWithTasks(final ListEntity entity, final List<TaskInformation> tasks) {
        return new ListInformation(entity.id, entity.name, entity.categoryId, tasks);
    }

    public ListInformation toListWithTasksModel(final ListsWithTasks entity) {
        final List<TaskInformation> listOfTasks = new ArrayList<>();

        for (final TaskEntity taskEntity : entity.tasks) {
            listOfTasks.add(new TaskInformation(taskEntity.id, taskEntity.name, taskEntity.description,
                    taskEntity.priority, TaskStatus.getStatus(taskEntity.status)));
        }

        return this.toModelWithTasks(entity.list, listOfTasks);
    }
}

File: app/src/main/java/com/example/pam_project/database/lists/ListMapper.kt
package com.example.pam_project.database.lists

import com.example.pam_project.database.relationships.ListsWithTasks
import com.example.pam_project.features.lists.list.ListInformation
import com.example.pam_project.features.tasks.list.TaskInformation
import com.example.pam_project.utils.constants.TaskStatus
import java.util.*

class ListMapper {
    fun toModel(entity: ListEntity?): ListInformation {
        return ListInformation(id = entity!!.id, title = entity.name, categoryId = entity.categoryId)
    }

    private fun toModelWithTasks(entity: ListEntity?, tasks: List<TaskInformation>?): ListInformation {
        return ListInformation(id = entity!!.id, title = entity.name, categoryId = entity.categoryId, tasks = tasks)
    }

    fun toListWithTasksModel(entity: ListsWithTasks?): ListInformation {
        val listOfTasks: MutableList<TaskInformation> = ArrayList()
        for (taskEntity in entity!!.tasks) {
            listOfTasks.add(TaskInformation(
                id = taskEntity.id,
                title = taskEntity.name,
                description = taskEntity.description,
                isUrgent = taskEntity.priority,
                status = TaskStatus.getStatus(taskEntity.status)
            ))
        }
        return toModelWithTasks(entity.list, listOfTasks)
    }
}